File: app/src/main/java/com/galib/natorepbs2/viewmodel/AchievementViewModel.java
package com.galib.natorepbs2.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.galib.natorepbs2.db.Achievement;
import com.galib.natorepbs2.db.NPBS2Repository;

import java.util.ArrayList;
import java.util.List;

public class AchievementViewModel extends AndroidViewModel {
    private NPBS2Repository mRepository;

    public AchievementViewModel(Application application) {
        super(application);
        mRepository = new NPBS2Repository(application);
    }

    public LiveData<List<Achievement>> getAllAchievement() {
        return mRepository.getAllAchievement();
    }

    public void insertFromArray(List<List<String>> trtd) {
        List<Achievement> achievements = new ArrayList<>();
        for(int i =0; i<trtd.size(); i++){
            if(i == 1) continue;
            achievements.add(new Achievement(trtd.get(i).get(0), trtd.get(i).get(1),
                    trtd.get(i).get(2), trtd.get(i).get(3), trtd.get(i).get(4), i));
        }
        mRepository.insertAchievementAll(achievements);
    }

    public void deleteAllAchievements() {
        mRepository.deleteAllAchievements();
    }
}

File: app/src/main/java/com/galib/natorepbs2/viewmodel/AchievementViewModel.kt
package com.galib.natorepbs2.viewmodel

import androidx.lifecycle.*
import com.galib.natorepbs2.models.Achievement
import com.galib.natorepbs2.db.NPBS2Repository
import kotlinx.coroutines.launch

class AchievementViewModel(private val mRepository: NPBS2Repository) : ViewModel() {

    val allAchievement: LiveData<List<Achievement>>
        get() = mRepository.allAchievement.asLiveData()

}

class AchievementViewModelFactory(private val repository: NPBS2Repository) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(AchievementViewModel::class.java)) {
            @Suppress("UNCHECKED_CAST")
            return AchievementViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}